### PR TITLE
fix: removed hard coded role and added a prop

### PR DESCRIPTION
**Pull request by yasgreen93**
Related to [TEC-1053](https://jira.economist.com/browse/TEC-1053).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -48,23 +50,26 @@ export default function Button(props) {
     Reflect.deleteProperty(linkProps, 'icon');
   }
 
-  linkProps.role = 'button';
+  if (props.buttonRole) {
+    linkProps.role = props.buttonRole;
+  }
+  const LinkComponent = props.linkComponent || 'a';
```
Comment (fabiosantoscode) [inline src/index.js]: Use `LinkComponent` in caps, as it's a React component.

File: src/index.js [hunk 2]
```diff
@@ -74,4 +79,6 @@ Button.propTypes = {
   icon: React.PropTypes.shape(Icon.propTypes),
   // i13n genuinely takes any object
   i13nModel: React.PropTypes.object, // eslint-disable-line react/forbid-prop-types
+  buttonRole: React.PropTypes.string,
+  linkComponent: React.PropTypes.object, // eslint-disable-line react/forbid-prop-types
```
Comment (fabiosantoscode) [inline src/index.js]: And here, it can be a string or a function as well. It may create some more warnings on the console (ew!)

**Comment by fabiosantoscode** [comment]
LGTM, @ricardofbarros up to you now

**Comment by ricardofbarros** [comment]
LLTM (looks lovely to me) @yasgreen93 :rainbow:

**Pull request merged**